src/main/scala/java/security/SecureRandom.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * scalajs-fake-insecure-java-securerandom (https://github.com/scala-js/scala-js-fake-insecure-java-securerandom)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package java.security
+
+import scala.scalajs.js
+import scala.scalajs.js.typedarray._
+
+// The seed in java.util.Random will be unused, so set to 0L instead of having to generate one
+class SecureRandom() extends java.util.Random(0L) {
+  // Make sure to resolve the appropriate function no later than the first instantiation
+  private val getRandomValuesFun = SecureRandom.getRandomValuesFun
+
+  // setSeed has no effect
+  override def setSeed(x: Long): Unit = ()
```

Comment:
This is odd. Reading the documentation, I would expect the following to end up being deterministic:

```
val x = new SecureRandom
x.setSeed(0)  // seeds x, doc says x will be seeded lazily
x.nextInt()
```

However, experimentally, it seems that the result is not deterministic. Hence, this implementation is OK.

Maybe comment on that?

Reply:
I'll add a comment.

The idea is that, for cryptographically secure RNG, giving a seed can only ever *increase* the entropy. It is never allowed to decrease it. Given that we don't have access to an API to strengthen the entropy of the underlying RNG, it's fine to ignore it instead.

Reply:
Yes, that makes sense as soon as the SecureRandom is seeded. But the documentation is very clear that seeding only happens once a random value is actually required. But yes, maybe the term "seeing" here is a bit overloaded.